.github/workflows/deprecations.yml

```diff
@@ -0,0 +1,45 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Crawl Deprecations
+
+on:
+  push:
+    branches:
+      - setup-git-action
+
+jobs:
+  crawl-and-update:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [12.x]
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          fetch-depth: 0
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: git fetch --no-tags --prune --depth=5 origin master
+      - name: crawl
+        run: npx deprecation-crawler --path ./deprecation-crawler.config.github-action.json --tag master
```

Comment:
How does this work?
I think it behaves as follows, but I'm probably wrong and that's why I'm confused now 😅

This will run the crawler against the master branch.
But AFAIK the master branch doesn't have the changes made in the PR, thus it won't find new deprecations (or these are always the deprecations that were introduced in a previous PR).
This will also commit the changes directly into `master`. Most of the projects don't allow that.

Reply:
It starts (should start) when we push to master. The crawler does its job (we should remove git commit step) and the next action pushes it to a branch e.g. master. but in our case a testing branch